redwood-protocol-guest/src/commonMain/kotlin/app/cash/redwood/protocol/guest/ProtocolWidgetSystemFactory.kt

```diff
@@ -15,12 +15,23 @@
  */
 package app.cash.redwood.protocol.guest
 
+import app.cash.redwood.Modifier
+import app.cash.redwood.RedwoodCodegenApi
+import app.cash.redwood.protocol.ModifierTag
 import app.cash.redwood.widget.WidgetSystem
+import kotlinx.serialization.KSerializer
 
 public interface ProtocolWidgetSystemFactory {
   /** Create a new [WidgetSystem] connected to a host via [guestAdapter]. */
   public fun create(
     guestAdapter: GuestProtocolAdapter,
     mismatchHandler: ProtocolMismatchHandler = ProtocolMismatchHandler.Throwing,
   ): WidgetSystem<Unit>
+
+  @RedwoodCodegenApi
+  public fun modifierTag(element: Modifier.Element): ModifierTag
```

Comment:
Tempted to do one function that returns a `Pair<ModifierTag, KSerializer<T>?>` instead of two functions. The only hazard of doing one function is I’d like to avoid allocating a `Pair` if possible.

Reply:
We generate the `KSerializer` for every modifier. We can do two things:

 1. Switch to subtyping `SerializationStrategy` in the generated type rather than `KSerializer`. We generate custom serializers because we only want the serialization code, so we should only be implementing the serialization interface (no clue why I didn't do this initially).
 2. Define a `ProtocolSerializationStrategy` interface which adds `val tag: ModifierTag` to `SerializationStrategy` and switch our codegen to use that. This eliminates the need to do a double, type-based lookup.

Reply:
I like both of these ideas. I’ll add a commit to this PR to do that!

Reply:
Done. Unfortunately I couldn’t do it exactly as suggested 'cause we don’t have a serializer for stateless modifiers. So instead it’s a `Pair<ModifierTag, SerializationStrategy>`. Seems to work okay and read okay.